libcaf_core/caf/flow/observable.hpp

```diff
@@ -330,6 +330,11 @@ class observable_def {
     return add_step(step::on_error_complete<output_type>{});
   }
 
+  auto on_error_return_item(output_type&& f) {
+    return add_step(
+      step::on_error_return_item<output_type>{std::forward<output_type>(f)});
+  }
+
```

Comment:
Could it be that you mistook this for a universal ref? The `output_type` is a known type here, so this will always be an rvalue ref. Please just pass it by value and then move into place.

```suggestion
  auto on_error_return_item(output_type item) {
    return add_step(
      step::on_error_return_item<output_type>{std::move(item)});
  }

```